chain-registry/src/utils.rs

```diff
@@ -0,0 +1,62 @@
+use crate::error::RegistryError;
+use async_trait::async_trait;
+use http::uri::Builder;
+use serde::de::DeserializeOwned;
+
+pub const PROTOCOL: &str = "https";
+pub const HOST: &str = "raw.githubusercontent.com";
+pub const REGISTRY_PATH: &str = "/cosmos/chain-registry";
+pub const REF: &str = "ffdfbff3a21c7d2dbaac6f1e4730f47063f4bd72";
+
+pub const TEST_CHAINS: &[&str] = &["cosmoshub", "juno", "evmos", "osmosis", "regen"];
+
+/// `Fetchable` represents the basic expectations for external data or resources that
+/// can be fetched.
+#[async_trait]
+pub trait Fetchable
+where
+    Self: DeserializeOwned,
+{
+    /// The file name of the fetchable resource.
+    fn file_name() -> String;
+
+    /// Fetches the fetchable resource.
+    // The default implementation fetches config data from a chain registry. This
+    // should be overridden if you're looking to fetch any other type of resource.
+    async fn fetch(chain_name: String) -> Result<Self, RegistryError> {
```

Comment:
No need for the `chain_name` to be owned, let's just use a `&str`:

```suggestion
    async fn fetch(chain_name: &str) -> Result<Self, RegistryError> {
```